sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -140,21 +143,37 @@ public VectorizedColumnReader(
     this.writerVersion = writerVersion;
   }
 
-  private boolean isLazyDecodingSupported(PrimitiveType.PrimitiveTypeName typeName) {
-    return switch (typeName) {
-      case INT32 ->
-        !(logicalTypeAnnotation instanceof DateLogicalTypeAnnotation) ||
-          "CORRECTED".equals(datetimeRebaseMode);
-      case INT64 -> {
-        if (updaterFactory.isTimestampTypeMatched(TimeUnit.MICROS)) {
-          yield "CORRECTED".equals(datetimeRebaseMode);
-        } else {
-          yield !updaterFactory.isTimestampTypeMatched(TimeUnit.MILLIS);
-        }
+  private boolean isLazyDecodingSupported(PrimitiveType.PrimitiveTypeName typeName,
+                                          DataType sparkType) {
+    boolean isSupported = false;
+    // Don't use lazy dictionary decoding if the column needs extra processing: upcasting or date
+    // rebasing.
+    switch (typeName) {
+      case INT32: {
```

Comment:
If it is necessary to switch back to using regular switch statements, the `{}` here is not needed

Reply:
This scope is needed to declare variables `needsUpcast` and `needsRebase` in both INT32 and INT64 branches without conflict. The alternative would be to declare the variables before the switch statement but that would be awkward since these only make sense in some of the switch branches.

Reply:
OK